src/main/java/nl/nn/testtool/filter/View.java

```diff
@@ -36,86 +40,114 @@
-		boolean match = false;
-		if (checkpointMatchers == null) {
-			match = true;
-		}
-		for (int i = 0; !match && i < checkpointMatchers.size(); i++) {
-			CheckpointMatcher checkpointMatcher = checkpointMatchers.get(i);
-			match = checkpointMatcher.match(report, checkpoint);
-		}
-		return match;
-	}
-
-	public String toString() {
-		return getName();
-	}
+    protected String name;
+    protected String nodeLinkStrategy;
+    private @Setter @Getter @Inject @Autowired LogStorage debugStorage;
+    private @Setter @Getter @Inject @Autowired List<String> metadataNames;
+    private Map<String, String> metadataFilter;
+    private List<CheckpointMatcher> checkpointMatchers;
+    private BeanParent beanParent;
+    private Echo2Application echo2Application;
+    private @Inject @Autowired MetadataExtractor metadataExtractor;
+
+    protected enum NodeLinkStrategy {
+        PATH,
+        CHECKPOINT_NUMBER,
+        NONE
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setNodeLinkStrategy(String nodeLinkStrategy) {
+        this.nodeLinkStrategy = nodeLinkStrategy;
+    }
+
+    public String getNodeLinkStrategy() {
+        if (nodeLinkStrategy == null) {
+            return NodeLinkStrategy.PATH.toString();
+        }
+        return nodeLinkStrategy;
+    }
+
+    public void setMetadataFilter(Map<String, String> metadataFilter) {
+        this.metadataFilter = metadataFilter;
+    }
+
+    public Map<String, String> getMetadataFilter() {
+        return metadataFilter;
+    }
+
+    public void setCheckpointMatchers(List<CheckpointMatcher> checkpointMatchers) {
+        this.checkpointMatchers = checkpointMatchers;
+    }
+
+    /**
+     * @see nl.nn.testtool.echo2.Echo2Application#initBean()
+     */
+    public void initBean(BeanParent beanParent) {
+        this.beanParent = beanParent;
+        this.echo2Application = Echo2Application.getEcho2Application(beanParent, this);
+    }
+
+    public BeanParent getBeanParent() {
+        return beanParent;
+    }
+
+    public Echo2Application getEcho2Application() {
+        return echo2Application;
+    }
+
+    public String isOpenReportAllowed(Object StorageId) {
+        return ReportsComponent.OPEN_REPORT_ALLOWED;
+    }
+
+    public boolean showCheckpoint(Report report, Checkpoint checkpoint) {
+        boolean match = false;
+        if (checkpointMatchers == null) {
+            match = true;
+        }
+        for (int i = 0; !match && i < checkpointMatchers.size(); i++) {
+            CheckpointMatcher checkpointMatcher = checkpointMatchers.get(i);
+            match = checkpointMatcher.match(report, checkpoint);
+        }
+        return match;
+    }
+
+    public String toString() {
+        return getName();
+    }
+
+    public Map<String, Object> toHashMap(boolean isDefaultView) {
+        Map<String, String> metadataTypes = new HashMap<>();
```

Comment:
Why call it toHashMap instead of toMap?

Reply:
Because Map is the interface type and HashMap the implemented returned type, I think it makes sense to return the actual type instead of its supertype.

Reply:
I would say use interface types for your "API" / what you expose to the outside world and use a specific type internally which you can change when needed without breaking the contract of your method (as far as I know this is common practice)

Reply:
Fixed this